src/Symfony/Bundle/FrameworkBundle/Command/ServerRunCommand.php

```diff
@@ -154,6 +160,11 @@ private function createPhpProcessBuilder(InputInterface $input, OutputInterface
             return;
         }
 
-        return new ProcessBuilder(array($binary, '-S', $input->getArgument('address'), $router));
+        $address = $input->getArgument('address');
+        if (false === strpos($address, ':')) {
+            $address = $address.':'.$input->getOption('port');
+        }
```

Comment:
I think we should rather pass the built address here instead of doing this all again.